components/tikv_util/src/logger/mod.rs

```diff
@@ -31,13 +32,38 @@ pub fn init_log<D>(
     level: Level,
     use_async: bool,
     init_stdlog: bool,
+    mut disabled_targets: Vec<String>,
 ) -> Result<(), SetLoggerError>
 where
     D: Drain + Send + 'static,
     <D as Drain>::Err: std::fmt::Display,
 {
+    // Only for debug purpose, so use environment instead of configuration file.
+    if let Ok(extra_modules) = env::var("TIKV_DISABLE_LOG_TARGETS") {
+        disabled_targets.extend(extra_modules.split(',').map(ToOwned::to_owned));
+    }
+
+    let filtered = drain.filter(move |record| {
+        if disabled_targets.is_empty() {
+            // The format of the returned value from module() would like this:
+            // ```
+            //  tikv::raftstore::store::fsm::store
+            //  tikv_util
+            //  tikv_util::config::check_data_dir
+            //  raft::raft
+            //  grpcio::log_util
+            //  ...
+            // ```
+            // Here get the highest level module name to check.
+            let module = record.module().splitn(2, "::").nth(0).unwrap();
+            disabled_targets.iter().any(|target| target == module)
```

Comment:
mostly fine. I'm only worried about the extra cost introduced by this precise blacklist checking, since in real world we won't need it.

Reply:
that's why I add the empty check. And equal maybe even faster than `start_with` due to the quick return of mismatch length (I don't take a look at the code, maybe wrong)